content/en/docs/architecture/bugzilla.md

```diff
@@ -0,0 +1,62 @@
+of checks:
+* If a bug with the id `123` exists
+* If repo has the [`is_open: true`][3] setting, the bug must not be closed
+* If the target release of the bugzilla bug matches the configured target release for the branch the PR points at. This mapping is
+  configured in the openshift/release repository via the `target_release` property. [Here][2] for example the `release-4.6` branch of the
+  openshift/ceph-csi repo is configured to have the `target_release` `OCS 4.6.0`
+* If the bug is in the correct state. Accepted states are configured in the openshift/release repository through the `valid_states`
+  property. An example where the `valid_states` are configured for all branches of the openshift/ceph-csi repo can be found [here][1]
+* If the repository has `dependent_bugs` and `dependent_bug_states` configured, those dependent bugs must be in one of the configured states
+
+If all of these checks succeed, the Bugzilla plugin will link the Pull Request in Bugzilla. If the configuration for the given
+github org, branch or repository contains the `state_after_validation` setting, the Bugzilla bug will be moved to that state
+after all validations succeeded.
+
+Similarly, the `state_after_close` and `state_after_merge` setting will cause the Bugzilla bug to be moved into the referenced
+state after close/merge.
+
+The Bugzilla automation will always comment on GitHub and mention the validations it ran. If any of the validations had an error,
+it will be there. Also, it adds `bugzilla/valid-bug` or `bugzilla/invalid-bug` label to the Pull Requests. Pull Requests with the
+`bugzilla/invalid-bug` label are never allowed to merge. Release branches in OCP usually require the `bugzilla/valid-bug` label.
+
+After a PR that references a Bugzilla bug merges, the bug will be put into the state that is configured as [`state_after_merge`][4].
+This is commonly used to trigger QE verification.
+
+You can find a detailed list of the bugzilla plugin settings for your repo in the [Prow Plugins][5] page by first selecting your
+repository in the top left and then clicking on `details` of the Bugzilla plugin.
+
+## Cherry Picking/Backporting
+
+The Bugzilla automation can also assist in the backporting process by integrating with the cherry picking automation. After an
+automated cherry-pick was created for a Pull Request that referenced a Bugzilla bug, the automation will clone the original bug
+with the new target release and retitle the PR to correctly reference that new Bug. The new bug will also be marked as dependent on the original bug.
+
+## Integration with QE verification
+
+Changes in OCP always end up in a nightly release assembled by the release-controller. The release-controller will run
+a set of verification jobs and if those pass, the release is considered accepted. Once that happens, the release-controller
+will:
+* Find all BZs associated with a change in the given release
```

Comment:
This may be a nuanced point but I believe it's better here to write again as if we are speaking to the user. We don't necessarily need to have a reference manual for how the release controller works. I'd perhaps write something more like this:

> Normally, QE will verify that a fix correctly resolves a Bugzilla bug once the bug transitions to the `ON_QA` state, which happens after the PR merges and the `release-controller` has assembled a new release payload containing the code.

> However, if a QE engineer is assigned to the Bugzilla bug, they will also get notified when a pull request is created to fix the bug. The QE engineer can leave an approving review on the pull request to indicate that the fix has been verified before the PR even merges. In this case, when the nightly release containing the fix is accepted, the Bugzilla bug automatically transitions past the `ON_QE` state and into `VERIFIED`.